src/models/mmt_encoder.py

```diff
@@ -0,0 +1,202 @@
+
+_NUM_OTHER_RELATIVE_IDS = 3
+
+
+class MmtEncoder(tf.keras.Model):
+  """Multimodal Transformer-based Encoder
+
+  This network implements a multimodal transformer-based encoder for 
+  image and language understanding. It includes the embedding lookups and 
+  relative transformer layers, but not the masked language model (mlm),
+  masked patch prediction (mpp), or classification task networks.
+
+  Args: refer to `MmtConfig` for more details.
+    We follow BERT implementation and use approximated gelu function for faster
+    TPU computation.
+    https://github.com/tensorflow/models/blob/2de518be2d6a6e3670b223a4582b1353538d3489/official/nlp/keras_nlp/encoders/bert_encoder.py#L26inner_activation
+    https://github.com/google/jax/issues/4428 
+  """
+
+  def __init__(self,
+               vocab_size: int,
+               segment_vocab_size: int = 16,
+               embedding_size: int = None,
+               hidden_size: int = 768,
+               num_hidden_layers: int = 12,
+               num_attention_heads: int = 12,
+               intermediate_size: int = 3072,
+               inner_activation=lambda x: tf.keras.activations.gelu(
+                   x, approximate=True),
+               hidden_dropout_prob: float = 0.1, 
+               attention_probs_dropout_prob: float = 0.1,
+               max_absolute_position_embeddings: Optional[int] = None,
+               relative_vocab_size: int = 32,
+               relative_pos_max_distance: int = 12,
+               initializer_range: float = 0.02,
+               use_pre_activation_order: bool = False,
+               use_one_hot_lookup: bool = True,
+               name: str = "mmt_encoder",
+               **kwargs):
+
+     super(MmtEncoder, self).__init__(name=name, **kwargs)
+
+     if relative_vocab_size is None:
+       if relative_pos_max_distance != 0:
+         raise ValueError(
+             '`relative_pos_max_distance` must be 0 when `relative_vocab_size` '
+             'is None.')
+     elif relative_vocab_size < (
+         etc_feature_utils.RelativePositionGenerator(
+             relative_pos_max_distance).relative_vocab_size +
+             _NUM_OTHER_RELATIVE_IDS):
+       raise ValueError(
+           f'`relative_vocab_size` ({relative_vocab_size}) too small for '
+           f'`relative_pos_max_distance` ({relative_pos_max_distance}')
+
+     if embedding_size is None:
+       embedding_size = hidden_size
+
+     activation = tf.keras.activations.get(inner_activation)
+     initializer = tf.keras.initializers.TruncatedNormal(
+         stddev=initializer_range)
+     initializer = tf.keras.initializers.get(initializer)
+
+     self._word_embedding_layer = etc_layers.EmbeddingLookup(
+         vocab_size=vocab_size,
+         embedding_size=embedding_size,
+         projection_size=hidden_size,
+         initializer_range=initializer_range,
+         name='word_embeddings')
+
+     if max_absolute_position_embeddings is not None:
```

Comment:
nit: for readability, I would suggest not negating the conditions unless necessary. So, just have this condition as "is None", and flip the "if" and the "else" portions below.